pkg/cache/scheduler/snapshot.go

```diff
@@ -52,14 +52,31 @@ func (s *Snapshot) RemoveWorkload(wl *workload.Info) {
 	cq.RemoveUsage(wl.Usage())
 }
 
-// AddWorkload adds a workload from its corresponding ClusterQueue and
+// AddWorkload adds a workload to its corresponding ClusterQueue and
 // updates resource usage.
 func (s *Snapshot) AddWorkload(wl *workload.Info) {
 	cq := s.ClusterQueue(wl.ClusterQueue)
 	cq.Workloads[workload.Key(wl.Obj)] = wl
 	cq.AddUsage(wl.Usage())
 }
 
+// SimulateWorkloadRemoval modifies the snapshot by removing the usage
+// corresponding to the list of workloads from workloads' respective
+// ClusterQueues. It returns a function which can be used to restore
+// this usage.
+func (s *Snapshot) SimulateWorkloadRemoval(workloads []*workload.Info) func() {
+	for _, w := range workloads {
```

Comment:
I think it is safer to revert this code. 
```
	usage := make([]workload.Usage, 0, len(workloads))
	for _, w := range workloads {
		usage = append(usage, w.Usage())
	}
```
I synced with @gabesaba it is better to revert, because otherwise some controller doing eviction could change `w.Usage()` in the meanwhile. as w is a pointer